cmd/nerdctl/image_prune.go

```diff
@@ -97,22 +99,39 @@ func imagePrune(ctx context.Context, cmd *cobra.Command, client *containerd.Clie
 	if err != nil {
 		return err
 	}
-	containerList, err := containerStore.List(ctx)
-	if err != nil {
-		return err
-	}
-	usedImages := make(map[string]struct{})
-	for _, container := range containerList {
-		usedImages[container.Image] = struct{}{}
-	}
 
 	delOpts := []images.DeleteOpt{images.SynchronousDelete()}
 	removedImages := make(map[string][]digest.Digest)
-	for _, image := range imageList {
-		if _, ok := usedImages[image.Name]; ok {
-			continue
+	var filteredImages []images.Image
+
+	if all {
+		containerList, err := containerStore.List(ctx)
+		if err != nil {
+			return err
 		}
+		usedImages := make(map[string]struct{})
+		for _, container := range containerList {
+			usedImages[container.Image] = struct{}{}
+		}
+
+		for _, image := range imageList {
+			if _, ok := usedImages[image.Name]; ok {
+				continue
+			}
+
+			filteredImages = append(filteredImages, image)
+		}
+	} else {
+		for _, image := range imageList {
+			_, tag := imgutil.ParseRepoTag(image.Name)
+
+			if tag == "" {
+				filteredImages = append(filteredImages, image)
+			}
+		}
```

Comment:
nit: after either this PR or #1931 is merged, maybe good to rebase the other and use the existing `filterDangling` (https://github.com/containerd/nerdctl/pull/1931#discussion_r1088125051).

Reply:
Yep. I'm waiting to do this.

Reply:
Rebased. @djdongjin